src/sentry/utils/performance_issues/performance_detection.py

```diff
@@ -907,33 +911,48 @@ def _validate_and_store_performance_problem(self):
         exceeds_count_threshold = len(self.consecutive_db_spans) >= self.settings.get(
             "consecutive_count_threshold"
         )
-        if exceeds_count_threshold and self._are_db_spans_independant(self.consecutive_db_spans):
+        independant_db_spans = self._find_independant_spans(self.consecutive_db_spans)
+        exceeds_duration_threshold = self._sum_span_duration(
+            independant_db_spans
+        ) > self.settings.get("duration_threshold")
+        if exceeds_count_threshold and exceeds_duration_threshold:
             self._store_performance_problem()
 
     def _store_performance_problem(self) -> None:
         fingerprint = self._fingerprint()
+        offender_span_ids = list(
+            map(lambda span: span.get("span_id", None), self.consecutive_db_spans)
+        )
```

Comment:
This should fix the issue where the `_pi_consecutive_db` tag is not set correctly.